### ISSUE TITLE
Fishing in Dalaran

### ISSUE THREAD

**Issue by TecDian** [TrinityCore/TrinityCore#21828]
**Description:**
Fishing in Dalaran gives the loot from Crystalsong Forest. This makes it impossible to finish two Dalaran fishing quests and the coin master achievements.

**Expected behaviour:**

There should be the fishing loot according to the Zone Dalaran.

**Steps to reproduce the problem:**

1. Just use your fishing rod there.

**Branch(es):** 3.3.5

**TC rev. hash/commit:** 3cd7851cd95d
TrinityCore rev. 3cd7851cd95d 2018-04-09 16:53:41 +0200 (3.3.5 branch) (Win64, Release, Static)
 Using SSL version: OpenSSL 1.0.2l 25 May 2017 (library: OpenSSL 1.0.2l 25 May 2017)
 Using Boost version: 1.65.0
 Using MySQL version: 5.6.37
 Using CMake version: 3.9.3
 Compiled on: Windows 10.0.16299
 Automatic database updates are disabled for all databases!
 Worldserver listening connections on port 8085
 Realmlist (Realm Id: 1) configured in port 8085
 VMAPs Status: Enabled. LineOfSight: 1, getHeight: 1, indoorCheck: 1
 MMAPs status: Enabled
 maps directory located in ./maps. Total size: 253388027 bytes
 vmaps directory located in ./vmaps. Total size: 592050936 bytes
 mmaps directory located in ./mmaps. Total size: 2152621724 bytes
 Using deDE DBC Locale as default. All available DBC locales: deDE
 Using World DB: TDB 335.64

**Operating system:** Windows 10

**Comment by Regrad** [comment]
Confirm

**Comment by Regrad** [comment]
This bug come with one of this commits:
https://github.com/TrinityCore/TrinityCore/commit/b9c6bbb51df0ed10f74617482740fedb91ff5366
or
https://github.com/TrinityCore/TrinityCore/commit/2dadbda24ac1c2e6ff3dad650c7e25f4b3322342

**Comment by TecDian** [comment]
Yes, it seems so. It's about the z-coordinate not propperly dealt with. Label Comp-Database ist not correct, it's core related.

**Comment by Killyana** [comment]
On dalaran fountain: GroundZ: 146.405167 FloorZ: 644.303223 Have height data (Map: 1 VMap: 1 MMap: 1)
[Liquid level: 646.380371, ground: 644.303223, type: 13, flags 1, status: 2.

On the river bellow dalaran: GroundZ: 146.386154 FloorZ: 146.386154 Have height data (Map: 1 VMap: 1 MMap: 1)
Liquid level: 147.408890, ground: 146.386154, type: 5, flags 1, status: 4.

Looks correct for me

**Comment by TecDian** [comment]
As I said, the database is correct. Core logic isn't.
